cpp/oneapi/dal/algo/covariance/detail/compute_ops.cpp

```diff
@@ -23,12 +24,36 @@ namespace v1 {
 
 template <typename Policy, typename Float, typename Method, typename Task>
 struct compute_ops_dispatcher<Policy, Float, Method, Task> {
-    compute_result<Task> operator()(const Policy& policy,
+    compute_result<Task> operator()(const Policy& ctx,
                                     const descriptor_base<Task>& desc,
+                                    const compute_parameters<Task>& params,
                                     const compute_input<Task>& input) const {
+        return implementation(ctx, desc, params, input);
+    }
+
+    compute_parameters<Task> select_parameters(const Policy& ctx,
+                                               const descriptor_base<Task>& desc,
+                                               const compute_input<Task>& input) const {
+        using kernel_dispatcher_t = dal::backend::kernel_dispatcher< //
+            KERNEL_SINGLE_NODE_CPU(parameters::compute_parameters_cpu<Float, Method, Task>)>;
+        return kernel_dispatcher_t{}(ctx, desc, input);
+    }
+
+    compute_result<Task> operator()(const Policy& ctx,
+                                    const descriptor_base<Task>& desc,
+                                    const compute_input<Task>& input) const {
+        const auto params = select_parameters(ctx, desc, input);
+        return implementation(ctx, desc, params, input);
+    }
+
+private:
+    inline auto implementation(const Policy& ctx,
+                               const descriptor_base<Task>& desc,
+                               const compute_parameters<Task>& params,
+                               const compute_input<Task>& input) const {
         using kernel_dispatcher_t = dal::backend::kernel_dispatcher< //
```

Comment:
I see the same `//` to achieve a line break was already in the code ... hmm, is this really the best way to do it?